internal/lib/container_server.go

```diff
@@ -319,6 +319,9 @@ func (c *ContainerServer) LoadSandbox(ctx context.Context, id string) (sb *sandb
 		return sb, err
 	}
 
+	// We should restore the infraContainer to the container state store
+	c.AddInfraContainer(ctx, scontainer)
```

Comment:
Is it feasible to add test coverage for this change?

Reply:
@sohankunkerkar, I was wondering about this.

This seems like an obvious thing to do. So I wonder why we didn't in the past? Perhaps indeed a bug.

Reply:
Obvious thing to do, as in, load infra containers on restart.

Reply:
> Is it feasible to add test coverage for this change?

can we add a unit test to check if the infraContainer is loaded?  like this:
https://github.com/cri-o/cri-o/pull/7726/files#diff-5df3a97ff31edf6084a160b044038d23d5ef6d2ab76f5ad0e5b57d059c37bdbcR160